mutable_tree_test.go

```diff
@@ -29,15 +30,65 @@ var (
 	tVal2 = []byte("v2")
 )
 
-func setupMutableTree(t *testing.T) *MutableTree {
+func setupMutableTree(t *testing.T, skipFastStorageUpgrade bool) *MutableTree {
 	memDB := db.NewMemDB()
-	tree, err := NewMutableTree(memDB, 0, false)
+	tree, err := NewMutableTree(memDB, 0, skipFastStorageUpgrade)
 	require.NoError(t, err)
 	return tree
 }
 
+// TestIterateConcurrency throws "fatal error: concurrent map writes" when fast node is enable
+func TestIterateConcurrency(t *testing.T) {
+	tree := setupMutableTree(t, true)
+
+	for i := 0; i < 100; i++ {
+		go func() {
```

Comment:
May I ask if need use local scope `i` to avoid data race, and [wait](https://github.com/mmsqe/iavl/blob/b1f03a903ac6c907a01a0eee042cdc8beda4c62b/mutable_tree_test.go#L40) until all set finish?

Reply:
If we wait then the tests will timeout; we just want to throw stuff at it enough that we are sure that it would fail.